library/std/src/panicking.rs

```diff
@@ -518,7 +518,7 @@ pub fn panicking() -> bool {
 }
 
 /// Entry point of panics from the libcore crate (`panic_impl` lang item).
-#[cfg(not(test))]
+#[cfg(not(any(test, doctest)))]
```

Comment:
I guess this is necessary because rustdoc unconditionally runs the lang item collection pass now.
We probably need to run benchmarks on this PR to check how it affects performance.